cdap-ui/app/cdap/components/PluginJSONCreator/Create/Content/index.tsx

```diff
@@ -73,23 +79,54 @@ const ContentView: React.FC<ICreateContext & WithStyles<typeof styles>> = ({
     return null;
   }
 
-  const Comp = STEPS[activeStep].component;
+  const renderContentPage = () => {
+    const Comp = STEPS[activeStep].component;
+    const CompLabel = STEPS[activeStep].label;
+    switch (CompLabel) {
+      case 'Basic Plugin Information':
+        return (
+          <PluginInfoContext.Provider value={pluginInfoContextValue}>
+            <Comp />
+          </PluginInfoContext.Provider>
+        );
+      case 'Configuration Groups':
+        return (
+          <ConfigurationGroupContext.Provider value={configuratioGroupContextValue}>
+            <WidgetContext.Provider value={widgetContextValue}>
+              <Comp />
+            </WidgetContext.Provider>
+          </ConfigurationGroupContext.Provider>
+        );
+      case 'Output':
+        return (
+          <OutputContext.Provider value={outputContextValue}>
+            <Comp />
+          </OutputContext.Provider>
+        );
+      case 'Filters':
+        return (
+          <FilterContext.Provider value={filterContextValue}>
+            <Comp />
+          </FilterContext.Provider>
+        );
+      default:
+        break;
```

Comment:
if you put this to the STEPS definition, then you won't need the switch at all. But just in case you do, this default cannot just `break;`. While ideally it shouldn't ever reach there, this will throw an error. You should return null in this case.

Reply:
addressed per our discussion.

Resolved.